### PR TITLE
[Botskills] Sanitize appSettings properties for all the commands

### PR DESCRIPTION
**Pull request by LCarbonellRod**
### Purpose

Depends on PR#XXXX

*What is the context of this pull request? Why is it being done?*
Botskills is running into retrocompatibility issues as the earlier versions was using the property `BotFrameworkSkills` with it properties inside, in uppercase. Now Botskills uses `botFrameworkSkills` in lowercase.

### Changes
*Are there any changes that need to be called out as significant or particularly difficult to grasp? (Include illustrative screenshots for context if applicable.)*
We are using the method `sanitizeAppSettingsProperties` for every time the AppSettings file is read. This method takes the `BotFrameworkSkills` property with it properties inside and change them to lowercase.

### Tests
*Is this covered by existing tests or new ones? If no, why not?*
We tested it manually and worked as expected. We added a unit test to check the `sanitizeAppSettingsProperties` method.

_BotFrameworkSkills in uppert case along with its properties_
image

_botFrameworkSkills after update using the sanitizeAppSettingsProperties method_
image

### Feature Plan
*Are there any remaining steps or dependencies before this issue can be fully resolved? If so, describe and link to any relevant pull requests or issues.*
\-

#### General
- [ ] I have commented my code, particularly in hard-to-understand areas	
- [x] I have added or updated the appropriate tests	
- [ ] I have updated related documentation

### CODE REVIEW

File: tools/botskills/src/functionality/listSkill.ts [hunk 1]
```diff
@@ -21,8 +22,9 @@ Please make sure to provide a valid path to your Assistant Skills configuration
 
                 return false;
             }
-            // Take VA Skills configurations
-            const assistantAppSettingsFile: IAppSetting = JSON.parse(readFileSync(configuration.appSettingsFile, 'UTF8'));
+            // Take VA Skills configurations            
+            const assistantAppSettingsFile: IAppSetting = JSON.parse(sanitizeAppSettingsProperties(configuration.appSettingsFile));
+
```
Comment (Batta32) [inline tools/botskills/src/functionality/listSkill.ts:27]: Remove this blank line!

Comment (LCarbonellRod) [reply tools/botskills/src/functionality/listSkill.ts]: Done!

File: tools/botskills/test/mocks/appsettings/appsettingsWithUppercase.json [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+        "key": ""
+    },
+    "BotFrameworkSkills": [
+        {
+            "Id": "testSkill",
+            "AppId": "",
+            "SkillEndpoint": "",
+            "Name": "",
+            "Description": ""
+        }
```
Comment (Batta32) [inline tools/botskills/test/mocks/appsettings/appsettingsWithUppercase.json:28]: Can you fill these values?

Comment (LCarbonellRod) [reply tools/botskills/test/mocks/appsettings/appsettingsWithUppercase.json:28]: I dind't fill them as it is a copy of other test appSettingsFile which doesn't has those values filled. I can fill the values but is it ok?

Regarding the first request change, I thought that I had to test the main commands where the util is used. However I will remove the tests to add one in the sanitation util tests.

Comment (Batta32) [reply tools/botskills/test/mocks/appsettings/appsettingsWithUppercase.json:28]: 1. About filling the properties, I guess that both ways are ok. So it's okay to be empty following what you said.
1. About the other, we are adding [unit tests](https://www.guru99.com/unit-testing-guide.html) where we are testing the individual units or components separately. In this case, we added the `sanitizationAppSettingsFile` method, and we need to test specifically that method. I understand what you mentioned, but the sanitization file is the responsible of that check.

File: tools/botskills/src/functionality/listSkill.ts [hunk 3]
```diff
@@ -21,8 +22,8 @@ Please make sure to provide a valid path to your Assistant Skills configuration
 
                 return false;
             }
-            // Take VA Skills configurations
-            const assistantAppSettingsFile: IAppSetting = JSON.parse(readFileSync(configuration.appSettingsFile, 'UTF8'));
+            // Take VA Skills configurations
```
Comment (Batta32) [inline tools/botskills/src/functionality/listSkill.ts:25]:
```suggestion
            // Take VA Skills configurations
```

File: tools/botskills/test/sanitization.test.js [hunk 4]
```diff
@@ -18,4 +19,17 @@ describe("The sanitization path util", function () {
...
+    describe("should return an appSettingsFile with the BotFrameworkSkills property and its values with the first letter in lowercase", function () {
+        it("when the first letter of the BotFrameworkSkills property and its values are in uppercase", async function() {
+            const pathToAppSettingsFile = join(__dirname, "mocks", "appsettings", "appsettingsWithUppercase.json");
+            const appSettingsObject = JSON.parse(sanitizeAppSettingsProperties(pathToAppSettingsFile));
+            strictEqual('botFrameworkSkills' in appSettingsObject, true);
+            strictEqual('id' in appSettingsObject.botFrameworkSkills[0], true);
+            strictEqual('name' in appSettingsObject.botFrameworkSkills[0], true);
+            strictEqual('appId' in appSettingsObject.botFrameworkSkills[0], true);
+            strictEqual('description' in appSettingsObject.botFrameworkSkills[0], true);
+            strictEqual('skillEndpoint' in appSettingsObject.botFrameworkSkills[0], true);
```
Comment (Batta32) [inline tools/botskills/test/sanitization.test.js:32]: I guess that you can change the `strictEqual` to the `ok` method.

* https://nodejs.org/api/assert.html#assert_assert_ok_value_message
* http://w3schools-fa.ir/nodejs/met_assert_ok.html#:~:text=Definition%20and%20Usage,and%20the%20program%20is%20terminated.

File: tools/botskills/test/sanitization.test.js [hunk 5]
```diff
@@ -18,4 +19,17 @@ describe("The sanitization path util", function () {
             strictEqual(sanitizePath(path), path.substring(0, path.length - 1));
         });
     })
+
+    describe("should return an appSettingsFile with the BotFrameworkSkills property and its values with the first letter in lowercase", function () {
+        it("when the first letter of the BotFrameworkSkills property and its values are in uppercase", async function() {
+            const pathToAppSettingsFile = join(__dirname, "mocks", "appsettings", "appsettingsWithUppercase.json");
+            const appSettingsObject = JSON.parse(sanitizeAppSettingsProperties(pathToAppSettingsFile));
```
Comment (Batta32) [inline tools/botskills/test/sanitization.test.js:26]:
```suggestion
            const appSettings = JSON.parse(sanitizeAppSettingsProperties(pathToAppSettingsFile));
```

File: tools/botskills/test/sanitization.test.js (hunk 4 above)
Comment (Batta32) [reply tools/botskills/test/sanitization.test.js:32]: Also, Can you confirm if the `in` operator is case sensitive or not? If its, this is okay.

Comment (LCarbonellRod) [reply tools/botskills/test/sanitization.test.js:32]: Yes I tested it and it is case sensitive. I'll add the `ok` method then.

File: tools/botskills/src/functionality/listSkill.ts (hunk 3 above)
Comment (Batta32) [reply tools/botskills/src/functionality/listSkill.ts:25]: You forgot this change! @LCarbonellRod

**Comment by LCarbonellRod** [comment]
> I double checked the changes and are okay, however, we can add the same behavior in the [migrate](https://github.com/southworks/botframework-solutions/blob/feature/southworks/botskills-sanitize-appsettings/tools/botskills/src/functionality/migrateSkill.ts#L42) command.

Done!

**Pull request merged**